bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -335,11 +351,11 @@ public int process(boolean shouldForceWrite) throws IOException {
                 lastLogMark.setCurLogMark(this.logId, this.lastFlushedPosition);
 
                 // Notify the waiters that the force write succeeded
-                for (QueueEntry e : this.forceWriteWaiters) {
-                    cbThreadPool.execute(e);
+                for (int i = 0; i < forceWriteWaiters.size(); i++) {
+                    cbThreadPool.execute((QueueEntry) forceWriteWaiters.get(i));
```

Comment:
it is not very good for type cast. but I think this question is related to the comment above - can we avoid using netty internal class. if we did avoid, can we make it type safe.

Reply:
Ya, the casts suck. I don't understand why the linter isn't complaining.